src/main/java/com/shapesecurity/salvation/Parser.java

```diff
@@ -595,6 +611,8 @@ private SourceExpression parseSourceExpression(Set<SeenStates> seenStates, boole
 						this.info(token, strictDynamicWarningMessage);
 					}
 					return new SchemeSource(token.value.substring(0, token.value.length() - 1));
+				} else if (token.value.equalsIgnoreCase("'unsafe-hashed-attributes'")) {
+					this.warn(token, "CSP specification renamed 'unsafe-hashed-attributes' to 'unsafe-hashes'.");
```

Comment:
Nit: This sentences should start with "The", or be rewritten to something like "'unsafe-hashed-attributes' has been renamed to 'unsafe-hashes'". Also ideally it would say when this happened (June 2018) or reference the PR where it happened (https://github.com/w3c/webappsec-csp/pull/311).